packages/block-editor/src/components/block-navigation/index.js

```diff
@@ -46,6 +46,7 @@ function BlockNavigationList( {
 							>
 								<BlockIcon icon={ blockType.icon } showColors />
 								{ blockType.title }
+								{ block.attributes.anchor && <span className="block-editor-block-navigation__item-anchor-name">{ block.attributes.anchor }</span> }
```

Comment:
What if a block uses the "anchor" attribute for something else? I wonder if we need a dedicated block API to show the "caption" of a block based on its attributes instead?

Thoughts @mtias @gziolo @mcsf

Reply:
I confirm that this implementation assumes a bit too much about the name of attribute. Enforcing that this attribute can’t be used for anything else might be too cumbersome. We should at least ensure that anchor is one the flag in the supports object defined for the block.


By the way, I know that the mobile part of project experiments with an API which creates a version of the block description which is used for accessibility purposes. It clearly shows that title and description are too generic when we have multiple instances of the same block in one document. We should seek to find a way to make it possible to offer more flexibility on how blocks can be described/announced depending on the context.

Reply:
I have refactored the code to check if the block supports the "anchor" feature.